chia/types/weight_proof.py

```diff
@@ -100,3 +103,63 @@ class WeightProof(Streamable):
     sub_epochs: List[SubEpochData]
     sub_epoch_segments: List[SubEpochChallengeSegment]  # sampled sub epoch
     recent_chain_data: List[HeaderBlock]
+
+
+@streamable
+@dataclass(frozen=True)
+class SubSlotDataV2(Streamable):
+    proof_of_space: Optional[ProofOfSpace]  # if challenge block
+    # VDF to signage point
+    cc_signage_point: Optional[VDFProof]  # if block
+    # VDF from signage to infusion point
+    cc_infusion_point: Optional[VDFProof]  # if block
+    signage_point_index: Optional[uint8]  # if block
+    cc_slot_end: Optional[VDFProof]  # if is end of slot
+    cc_sp_vdf_output: Optional[CompressedClassgroupElement]  # if is end of slot
+    cc_ip_vdf_output: Optional[CompressedClassgroupElement]  # if is end of slot
+    cc_slot_end_output: Optional[ClassgroupElement]  # if is end of slot
+    icc_infusion_point: Optional[VDFProof]
+    icc_ip_vdf_output: Optional[CompressedClassgroupElement]
+
+    icc_slot_end: Optional[VDFProof]
+    icc_slot_end_output: Optional[ClassgroupElement]
+    cc_sp_signature: Optional[G2Element]  # if challenge block (needed for ChallengeBlockInfo)
+    ip_iters: Optional[uint64]  # if block
+    total_iters: Optional[uint128]  # if block
```

Comment:
can we calculate this using slots and pospace?

Reply:
i dont think so because we only have pospace for the challenge blocks

Reply:
if we know the ip_iters we can calculate the total_iters right?

Reply:
sure, how ?

you just stared validating a new subepoch (yo dont know the total_iters of the previous block), what is the total iters of the first block given we have ip_iters x ?

maybe if we include the total_iters in the start of each sub_epoch sample

Reply:
Ah i see. Would it be worth doing that? It would save some data (128 bits for each of these).

Reply:
since we remove these field from all the parts where we dont do vdf validation i think it wont be significant i can do some runs tomorrow and find out exactly

Reply:
after checking it will save about 0.1 mb out of about 10 total i dont think we should change this